### ISSUE TITLE
Fails to build with ldc 1.28.0

### ISSUE THREAD

**Issue by tillea** [biod/BioD#55]
Hi,
there is a [bug report}(https://bugs.debian.org/998913) against the Debian packaged version of BioD.  We are currently packaging commit b8eecef as commited at 2019-11-20.  It would be great if you could tag your releases if these are closing issues like compatibility with certain compiler versions or get new features.
Kind regards, Andreas.

**Comment by pjotrp** [comment]
Thanks @tillea. I wonder if anyone is using BioD. Sambamba imported a subset of BioD, but no longer does that.

**Comment by tillea** [comment]
Hi Pjotr,

Am Mon, Nov 22, 2021 at 08:17:26AM -0800 schrieb Pjotr Prins:
> Thanks @tillea. I wonder if anyone is using BioD. Sambamba imported a subset of BioD, but no longer does that. 

The motivation to package BioD was only to get Sambamba packaged.  I have no idea whether there is any user of plain BioD (we simply can not know).  If you want to tell me that there is no sense to keep BioD in Debian I wonder whether it might be the best idea to simply orphan the project upstream.
Kind regards, Andreas.

**Comment by pjotrp** [comment]
Yes, I think we should orphan BioD until someone picks up the project again.

**Comment by tillea** [comment]
Hi,
unfortunately biod is specified in `meson.build` of sambamba.  I tried to [patch](https://salsa.debian.org/med-team/sambamba/-/blob/master/debian/patches/no_biod_needed.patch) it out but this ends up in a build failure:

            cd obj-x86_64-linux-gnu && LC_ALL=C.UTF-8 ninja -j4 -v
    [1/34] ldc2 -I=sambamba.p -I=. -I=.. -enable-color -wi -O -g -release -wi -O3 -release -enable-inlining -boundscheck=off -J../ -J. -makedeps=sambamba.p/sambamba_depth.d.o.deps -     of=sambamba.p/sambamba_depth.d.o -c ../sambamba/depth.d
    FAILED: sambamba.p/sambamba_depth.d.o
    ldc2 -I=sambamba.p -I=. -I=.. -enable-color -wi -O -g -release -wi -O3 -release -enable-inlining -boundscheck=off -J../ -J. -makedeps=sambamba.p/sambamba_depth.d.o.deps -of=sambamba.p/sambamba_depth.d.o -c ../sambamba/depth.d
    ../sambamba/depth.d(22): Error: module `reader` is in file 'bio/std/hts/bam/reader.d' which cannot be read
    import path[0] = sambamba.p
    import path[1] = .
    import path[2] = ..
    import path[3] = /usr/lib/ldc/x86_64-linux-gnu/include/d
    import path[4] = /usr/include/d
    [2/34] ldc2 -I=sambamba.p -I=. -I=.. -enable-color -wi -O -g -release -wi -O3 -release -enable-inlining -boundscheck=off -J../ -J. -makedeps=sambamba.p/sambamba_main.d.o.deps -      of=sambamba.p/sambamba_main.d.o -c ../sambamba/main.d
    FAILED: sambamba.p/sambamba_main.d.o
    ldc2 -I=sambamba.p -I=. -I=.. -enable-color -wi -O -g -release -wi -O3 -release -enable-inlining -boundscheck=off -J../ -J. -makedeps=sambamba.p/sambamba_main.d.o.deps -of=sambamba. p/sambamba_main.d.o -c ../sambamba/main.d
    ../utils/version_.d(6): Error: module `header` is in file 'bio/std/hts/sam/header.d' which cannot be read
    import path[0] = sambamba.p
    import path[1] = .
    import path[2] = ..
    import path[3] = /usr/lib/ldc/x86_64-linux-gnu/include/d
    import path[4] = /usr/include/d
    [3/34] ldc2 -I=sambamba.p -I=. -I=.. -enable-color -wi -O -g -release -wi -O3 -release -enable-inlining -boundscheck=off -J../ -J. -makedeps=sambamba.p/sambamba_flagstat.d.o.deps -  of=sambamba.p/sambamba_flagstat.d.o -c ../sambamba/flagstat.d
    FAILED: sambamba.p/sambamba_flagstat.d.o
    ldc2 -I=sambamba.p -I=. -I=.. -enable-color -wi -O -g -release -wi -O3 -release -enable-inlining -boundscheck=off -J../ -J. -makedeps=sambamba.p/sambamba_flagstat.d.o.deps -         of=sambamba.p/sambamba_flagstat.d.o -c ../sambamba/flagstat.d
    ../sambamba/flagstat.d(23): Error: module `reader` is in file 'bio/std/hts/bam/reader.d' which cannot be read
    import path[0] = sambamba.p
    import path[1] = .
    import path[2] = ..
    import path[3] = /usr/lib/ldc/x86_64-linux-gnu/include/d
    import path[4] = /usr/include/d
    [4/34] ldc2 -I=sambamba.p -I=. -I=.. -enable-color -wi -O -g -release -wi -O3 -release -enable-inlining -boundscheck=off -J../ -J. -makedeps=sambamba.p/sambamba_fixbins.d.o.deps -   of=sambamba.p/sambamba_fixbins.d.o -c ../sambamba/fixbins.d
    FAILED: sambamba.p/sambamba_fixbins.d.o
    ldc2 -I=sambamba.p -I=. -I=.. -enable-color -wi -O -g -release -wi -O3 -release -enable-inlining -boundscheck=off -J../ -J. -makedeps=sambamba.p/sambamba_fixbins.d.o.deps -          of=sambamba.p/sambamba_fixbins.d.o -c ../sambamba/fixbins.d
    ../sambamba/fixbins.d(25): Error: module `reader` is in file 'bio/std/hts/bam/reader.d' which cannot be read
    import path[0] = sambamba.p
    import path[1] = .
    import path[2] = ..
    import path[3] = /usr/lib/ldc/x86_64-linux-gnu/include/d
    import path[4] = /usr/include/d
    ninja: build stopped: subcommand failed.

It would be great if you could fully solve the biod dependency.

Kind regards, Andreas.

PS: Pjotr, the e-mail address I used before to reach you is bouncing currently

**Comment by pjotrp** [comment]
I don't think that dependency is necessary, but I'll check.

**Comment by tillea** [comment]
Am Thu, Dec 09, 2021 at 02:01:20AM -0800 schrieb Pjotr Prins:
> I don't think that dependency is necessary, but I'll check.

Thanks.  It would be great if you could tag a new point release without any reference to biod once you have successfully checked this.

**Comment by pjotrp** [comment]
See https://github.com/biod/sambamba/issues/482

**Comment by pjotrp** [comment]
I'll archive this repo soon.

**Comment by tillea** [comment]
Am Sun, Dec 26, 2021 at 08:29:21AM -0800 schrieb Pjotr Prins:
> I'll archive this repo soon.

Did you verified that sambamba builds without BioD?  It would be great to have a tagged release of sambamba that has no reference to BioD.
